README.md

```diff
@@ -9,6 +9,118 @@
 npm install screwdriver-scm-base
 ```
 
+## Interface
+This is a promise based interface for interacting with a source control management system
+
+### configure
+The `configure` function takes in an object and resets the configuration values
+
+### formatScmUrl
+Required parameters:
+
+| Parameter        | Type  |  Description |
+| :-------------   | :---- | :-------------|
+| scmUrl        | String | Scm Url to format |
+
+#### Expected Outcome
+A formatted scm url to be used as a unique key. This function is synchronous.
+
+### getPermissions
+Required parameters:
+
+| Parameter        | Type  |  Description |
+| :-------------   | :---- | :-------------|
+| config        | Object | Configuration Object |
+| config.scmUrl | String | The scmUrl to get permissions on |
+| config.token | String | The github token to check permissions on |
+
+#### Expected Outcome
+Permissions for a given token on a repository
```

Comment:
Should we say what object this should be?
`{ admin: true/false, push: true/false, pull: true/false }`